Sources/MlemMiddleware/FeedLoaders/Inbox/InboxFeedLoader/InboxItem.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  InboxItem.swift
+//  MlemMiddleware
+//
+//  Created by Eric Andrews on 2025-02-01.
+//
+
+public enum InboxItem: FeedLoadable, ReadableProviding, InboxIdentifiable {
+    public typealias FilterType = InboxItemFilterType
+    
+    case message(Message2)
+    case reply(Reply2)
+    
+    var baseValue: any FeedLoadable & ActorIdentifiable {
+        switch self {
+        case let .message(message2): message2
+        case let .reply(reply2): reply2
+        }
+    }
+    
+    public var read: Bool {
+        switch self {
+        case .message(let message2): message2.read
+        case .reply(let reply2): reply2.read
```

Comment:
imo this syntax is a little nicer

```suggestion
        case let .message(message2): message2.read
        case let .reply(reply2): reply2.read
```